Makefile

```diff
@@ -154,27 +157,76 @@ stylus_test_hostio-test_src       = $(call get_stylus_test_rust,hostio-test)
 
 stylus_test_wasms = $(stylus_test_keccak_wasm) $(stylus_test_keccak-100_wasm) $(stylus_test_fallible_wasm) $(stylus_test_storage_wasm) $(stylus_test_multicall_wasm) $(stylus_test_log_wasm) $(stylus_test_create_wasm) $(stylus_test_math_wasm) $(stylus_test_sdk-storage_wasm) $(stylus_test_erc20_wasm) $(stylus_test_read-return-data_wasm) $(stylus_test_evm-data_wasm) $(stylus_test_hostio-test_wasm) $(stylus_test_bfs:.b=.wasm)
 stylus_benchmarks = $(wildcard $(stylus_dir)/*.toml $(stylus_dir)/src/*.rs) $(stylus_test_wasms)
+CBROTLI_WASM_BUILD_ARGS ?=-d
+
+
+ESPRESSO_NETWORK_GO_VER ?= 0.0.36
+ESPRESSO_TAR = espresso-network-go-$(ESPRESSO_NETWORK_GO_VER).tar.gz
+ESPRESSO_URL = https://github.com/EspressoSystems/espresso-network-go/archive/refs/tags/v$(ESPRESSO_NETWORK_GO_VER).tar.gz
+ESPRESSO_DIR = espresso-network-go
 
-espresso_crypto_dir = ./espressocrypto/lib/espresso-crypto-helper
+# Download the tarball
+$(ESPRESSO_TAR):
+	curl -L -o $@ $(ESPRESSO_URL)
+
+# Extract into target directory (strip the top-level folder)
+$(ESPRESSO_DIR): $(ESPRESSO_TAR)
+	@echo "Extracting $(ESPRESSO_TAR) into $(ESPRESSO_DIR)/..."
+	rm -rf $(ESPRESSO_DIR)
+	mkdir -p $(ESPRESSO_DIR)
+	tar -xzf $(ESPRESSO_TAR) --strip-components=1 -C $(ESPRESSO_DIR)
+
+espresso_crypto_dir = $(ESPRESSO_DIR)/verification/rust
```

Comment:
Is there any obstacles using the command line to download the library? I can update it ASAP.

This command line can help us get rid of the tedious step of finding the arch and platform.

```
sudo go run github.com/EspressoSystems/espresso-network-go/download download
```

Reply:
okay let me try this and I will let you know how it goes!

Reply:
So sudo was messing up with a lot of things for me because my docker was not configured to run with sudo. I now made it a wildcard like this: COPY --from=node-builder /workspace/target/lib/libespresso_crypto_helper-*.so /usr/local/lib/ and that works!

Reply:
I think at some point it would be worth having a security discussion surrounding how we ask consumers of the crypto lib to download and run it. I would love for us to avoid an xz backdoor style situation with this library if at all possible.

Reply:
@zacshowa can you talk about this attack more? I dont know much about it